src/transports/dash/segment_parser.ts

```diff
@@ -26,64 +32,121 @@ import {
 import takeFirstSet from "../../utils/take_first_set";
 import {
   ISegmentParserArguments,
-  ISegmentParserObservable,
+  ISegmentParserResponseEvent,
+  ITransportRetryEvent,
 } from "../types";
 import isWEBMEmbeddedTrack from "./is_webm_embedded_track";
 import getISOBMFFTimingInfos from "./isobmff_timing_infos";
+import loadIndexes from "./load_indexes";
 
 export default function parser({ content,
                                  response,
-                                 init } : ISegmentParserArguments< Uint8Array |
-                                                                   ArrayBuffer |
-                                                                   null >
-) : ISegmentParserObservable< Uint8Array | ArrayBuffer > {
+                                 init,
+                                 scheduleRequest } : ISegmentParserArguments<Uint8Array |
+                                                                             ArrayBuffer |
+                                                                             null >
+) : Observable<ISegmentParserResponseEvent<Uint8Array|ArrayBuffer> |
+               ITransportRetryEvent> {
   const { period, representation, segment } = content;
   const { data, isChunked } = response;
   if (data == null) {
-    return observableOf({ chunkData: null,
-                          chunkInfos: null,
-                          chunkOffset: 0,
-                          appendWindow: [period.start, period.end] });
+    return observableOf({ type: "parser-response" as const,
+                          value: { chunkData: null,
+                                   chunkInfos: null,
+                                   chunkOffset: 0,
+                                   appendWindow: [period.start, period.end] } });
   }
-
   const chunkData = data instanceof Uint8Array ? data :
                                                  new Uint8Array(data);
 
-  const indexRange = segment.indexRange;
+  const { range } = segment;
   const isWEBM = isWEBMEmbeddedTrack(representation);
 
-  const nextSegments = isWEBM ?
-    getSegmentsFromCues(chunkData, 0) :
-    getSegmentsFromSidx(chunkData, Array.isArray(indexRange) ? indexRange[0] :
-                                                               0);
+  let nextSegments;
+  let indexes;
+
+  if (isWEBM) {
+    nextSegments = getSegmentsFromCues(chunkData, 0);
+  } else {
+    const indexOffset = Array.isArray(range) ? range[0] :
+                                               0;
+    const referencesFromSidx = getReferencesFromSidx(chunkData, indexOffset);
+    if (referencesFromSidx !== null) {
+      const [indexReferences, segmentReferences] = referencesFromSidx;
+      if (segmentReferences.length > 0) {
+        nextSegments = segmentReferences;
+      }
+      indexes = indexReferences;
```

Comment:
Uh, might be more readable to do just:
```
indexes = referencesFromSidx[0];
nextSegments = referencesFromSidx[1];
```
For those 5 lines.

Also, `indexes` is not a clear variable name.

Reply:
Done